platforms/core-configuration/model-core/src/test/groovy/org/gradle/api/internal/provider/CollectionPropertySpec.groovy

```diff
@@ -1291,6 +1301,17 @@ The value of this property is derived from: <source>""")
         property.explicit
     }
 
+    def "can add to explicit value with append"() {
```

Comment:
It would be good to have some test cases where the convention or explicit value changes after `append()` or `appendAll()` has been called. Same for when the property changes from using an explicit value to using the convention, and vice versa.

It might be good to have some test coverage for when calls to `add()` and `append()`, etc are mixed.

Same comment applies to the new methods on `MapProperty` too.